test/e2e/multicluster.go

```diff
@@ -82,6 +117,7 @@ func runSimpleWorkloadUpgradeFlowForBareMetal(test *framework.MulticlusterE2ETes
 }
 
 func runWorkloadClusterUpgradeFlowWithAPIForBareMetal(test *framework.MulticlusterE2ETest, filler ...api.ClusterConfigFiller) {
+	test.ManagementCluster.WithClusterConfig()
```

Comment:
I would remove this. This would probably be better called when the cluster test object is constructed like you have been already for example in `TestTinkerbellUpgrade126MulticlusterWorkloadClusterWorkerScaleupWithFluxAPI`

Reply:
Updated upgrade tests to pre-generate cluster config and removed the WithClusterConfig generation.